data_model/src/query.rs

```diff
@@ -1522,9 +1530,68 @@ pub mod block {
         }
     }
 
+    #[derive(
+        Default,
+        Debug,
+        Clone,
+        Copy,
+        PartialEq,
+        Eq,
+        Decode,
+        Encode,
+        Deserialize,
+        Serialize,
+        IntoSchema,
+        PartialOrd,
+        Ord,
+    )]
+    /// `FindAllBlockHeaders` Iroha Query will list all block headers
+    pub struct FindAllBlockHeaders;
+
+    impl Query for FindAllBlockHeaders {
+        type Output = Vec<BlockHeaderValue>;
+    }
+
+    impl FindAllBlockHeaders {
+        /// Construct [`FindAllBlockHeaders`].
+        pub fn new() -> Self {
+            Self {}
+        }
+    }
+
+    #[derive(
+        Debug,
+        Clone,
+        PartialEq,
+        Eq,
+        Decode,
+        Encode,
+        Deserialize,
+        Serialize,
+        IntoSchema,
+        PartialOrd,
+        Ord,
+    )]
+    /// `FindBlockHeaderByHash` Iroha Query will find block header by block hash
+    pub struct FindBlockHeaderByHash {
+        /// Block hash hash.
+        pub hash: EvaluatesTo<Hash>,
```

Comment:
```suggestion
        pub hash: EvaluatesTo<HashOf<VersionedCommittedBlock>>,
```

What about this? As I can see this is widely used pattern in our codebase

Reply:
Not sure, I just did it by analogy with `FindTransactionByHash`. I'll have a look if having typed hash is disadvantageous somehow.

Reply:
Ah, of course. We can't use `VersionedCommitedBlock` here, since it's defined in core.

Reply:
Indeed